frontend/public/components/environment.jsx

```diff
@@ -200,12 +305,14 @@ export const EnvironmentPage = connect(stateToProps)(
      */
     _reload() {
       const {rawEnvData} = this.props;
-      this.setState({
-        currentEnvVars: envVarsToArray(rawEnvData),
-        errorMessage: null,
-        success: null,
-        modified: false,
-        stale: false
+      this.setState((prevState) => {
+        return {
+          errorMessage: null,
+          success: null,
+          modified: false,
+          stale: false,
+          currentEnvVars: prevState.currentEnvVars.instantiateWithRawData(rawEnvData)
```

Comment:
Isn't this modifying the previous state object? (I think that should be avoided.)

Reply:
agreed!